components/ai_chat/core/browser/conversation_handler_unittest.cc

```diff
@@ -3623,4 +3625,158 @@ TEST_F(ConversationHandlerUnitTest_NoAssociatedContent,
+  ASSERT_EQ(1u, initial_content.size());
+  EXPECT_FALSE(initial_content[0]->conversation_turn_uuid.has_value());
+
+  // Associate content with turn
+  conversation_handler_->associated_content_manager()
+      ->AssociateUnsentContentWithTurn(turn);
+
+  // After association, GetAssociatedContent should have conversation_turn_uuid
+  // set
+  auto associated_content = conversation_handler_->associated_content_manager()
+                                ->GetAssociatedContent();
+  ASSERT_EQ(1u, associated_content.size());
+  EXPECT_TRUE(associated_content[0]->conversation_turn_uuid.has_value());
+  EXPECT_EQ("test-turn-uuid",
+            associated_content[0]->conversation_turn_uuid.value());
+  EXPECT_EQ(associated_content_->uuid(), associated_content[0]->uuid);
+
+  // GetCachedContentsMap should work without crashing and include the turn UUID
+  // as key
+  auto contents_map = conversation_handler_->associated_content_manager()
+                          ->GetCachedContentsMap();
+  EXPECT_TRUE(contents_map.contains("test-turn-uuid"));
+}
+
+TEST_F(ConversationHandlerUnitTest,
+       AssociateUnsentContentWithTurn_MultipleContent) {
+  // Add a second content delegate
+  auto second_content = std::make_unique<NiceMock<MockAssociatedContent>>();
+  conversation_handler_->associated_content_manager()->AddContent(
+      second_content.get(), false, false);
+
+  auto turn = mojom::ConversationTurn::New(
+      "test-turn-uuid", mojom::CharacterType::HUMAN, mojom::ActionType::QUERY,
+      "Test human message", std::nullopt, std::nullopt, std::nullopt,
+      base::Time::Now(), std::nullopt, std::nullopt, false, std::nullopt);
+
+  // Associate content with turn
+  conversation_handler_->associated_content_manager()
+      ->AssociateUnsentContentWithTurn(turn);
+
+  // Both content items should be associated with the turn
+  auto associated_content = conversation_handler_->associated_content_manager()
+                                ->GetAssociatedContent();
+  ASSERT_EQ(2u, associated_content.size());
+
+  // Both should have the same conversation_turn_uuid
+  EXPECT_TRUE(associated_content[0]->conversation_turn_uuid.has_value());
+  EXPECT_TRUE(associated_content[1]->conversation_turn_uuid.has_value());
+  EXPECT_EQ("test-turn-uuid",
+            associated_content[0]->conversation_turn_uuid.value());
+  EXPECT_EQ("test-turn-uuid",
+            associated_content[1]->conversation_turn_uuid.value());
+
+  // UUIDs should be match
+  EXPECT_EQ(associated_content_->uuid(), associated_content[0]->uuid);
+  EXPECT_EQ(second_content->uuid(), associated_content[1]->uuid);
+
+  // GetCachedContentsMap should work and have both content items under the same
+  // turn UUID
+  auto contents_map = conversation_handler_->associated_content_manager()
+                          ->GetCachedContentsMap();
+  EXPECT_TRUE(contents_map.contains("test-turn-uuid"));
+  EXPECT_EQ(2u, contents_map.at("test-turn-uuid").size());
```

Comment:
can we also associate a new content with a different turn so we can test if it is handled properly?

Reply:
Added `AssociateUnsentContentWithTurn_MultipleContent_MultipleTurns`